osu.Game/Skinning/IPooledSampleProvider.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using JetBrains.Annotations;
+using osu.Game.Audio;
+
+namespace osu.Game.Skinning
+{
+    public interface IPooledSampleProvider
+    {
+        [CanBeNull]
+        PoolableSkinnableSample GetPooledSample(ISampleInfo sampleInfo);
```

Comment:
is this realistically public API that will need xmldoc?

Reply:
this is the kind of interface i'd probably argue to be internal as i don't think rulesets should be providing their own pooling backend for samples.